File: cpp/0049.cpp
class Solution
{
public:
    vector<vector<string>> groupAnagrams(vector<string> &strs)
    {
        vector<vector<string>> res;
        map<string, vector<string>> table;

        for (auto str : strs)
        {
            string key = str;
            sort(key.begin(), key.end());
            table[key].push_back(str);
        }
        for (auto it : table)
        {
            res.push_back(it.second);
        }
        return res;
    }
};

File: py/0049.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = {}
        for s in strs:
            index = "".join((lambda x: (x.sort(), x)[1])(list(s)))
            if index not in dic:
                dic[index] = []
            dic[index].append(s)

        res = []
        for key in dic:
            res.append(dic[key])
        return res
